ArrayList/ArrayList.cs

```diff
@@ -1,8 +1,580 @@
 ﻿using System;
+using System.Text;
 
-namespace ArrayList
+namespace List
 {
-    public class ArrayList
+    public class ArrayList<T> where T : IComparable<T>
     {
+        public int Length
+        {
+            get
+            {
+                return _length;
+            }
+            private set
+            {
+                if (value >= 0)
+                {
+                    _length = value;
+                }
+                else if (value < 0)
+                {
+                    _length = 0;
+                }
+            }
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (index >= 0 && index < Length)
+                {
+                    return _array[index];
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+            set
+            {
+                if (index >= 0 && index < _array.Length)
```

Comment:
That's mean that user can do something like:
`lst.Add(1); lst.Add(2); lst[5] = 10;`
A bit freak, isn't it?

Reply:
is an implementation of the "Change by index" task.

Reply:
I mean that you can't use `_array.Length` here because you current array could have real length 10 items but explicitly added only 2. In this case you should block adding using index more than explicit added count.

Reply:
ok